### PR TITLE
feat: Adding UsageStatistics report when the form filler was used for the first time per JVM.

### PR DESCRIPTION
**Pull request by czp13**
## Description
I was thinking and could not find a better place to put this for now. My requirements were:
- Shall only run once per JVM (loading the app),
- The properties (version at least) should have been dynamically loaded,

So come up with this hybrid solution to have a `FormFillerStats` class, which is added to the `FormFiller` constructor, but will only run once (when the first time the FormFiller is used)

Other minor changes made in the `FormFiller` class:
- using final variables wherever it makes sense (constructor params),

Fixes # (issue)
- https://github.com/vaadin/form-filler-addon/issues/14
- https://github.com/vaadin/form-filler-addon/issues/31

Got some ideas from this PR:
- https://github.com/vaadin/hilla/pull/1055/files

And from thees ones:
- https://github.com/vaadin/testbench/pull/1587/files
- https://github.com/vaadin/sso-kit/pull/62

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [ ] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and correct misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria was created.

### CODE REVIEW

**Comment by Legioth** [comment]
> Shall only run once per JVM (loading the app)

No need for the complexity. The usage statistics handling already takes care of avoiding duplicate reports. Could probably just as well be a static init block in the `FormFiller` class.

**Review by mgarciavaadin — APPROVED** [review]
Stats seems to work fine! Just the injection from pom file for version property not working. Closing the PR and creating an issue to enhance in the future.

**Comment by czp13** [comment]
Thanks, also I wanted to fix what was asked by Leif, but I can do it in a separate PR.

File: src/main/java/org/vaadin/addons/ai/formfiller/FormFiller.java [hunk 1]
```diff
@@ -155,4 +156,20 @@ public ArrayList<String> getContextInstructions() {
...
+    /**
+     * Reports the usage statistics of the FormFiller component.
+     *
+     * The reporting of the usage statistics will only run once per JVM,
+     * when the FormFiller component is used for the first time.
+     */
+    private void reportUserStatistics() {
+        try {
+            // This will load the class if it exists, and so the static block will be executed and so the usage statistics will be reported.
+            // Nicer solution would have been if we could extend/overwrite the VaadinServiceInitListener somehow and execute the statistics there.
```
[the remark this answers is not in the document]
Comment (czp13) [reply src/main/java/org/vaadin/addons/ai/formfiller/FormFiller.java:176]: fixing here:
- https://github.com/vaadin/form-filler-addon/issues/77

**Comment by czp13** [comment]
> > Shall only run once per JVM (loading the app)
> 
> No need for complexity. The usage statistics handling already takes care of avoiding duplicate reports. Could probably just as well be a static init block in the `FormFiller` class.

This review comment will be fixed here:
- https://github.com/vaadin/form-filler-addon/issues/77

**Pull request merged**